CovertInfiltration/Src/CovertInfiltration/Classes/X2DownloadableContentInfo_CovertInfiltration.uc

```diff
@@ -816,6 +817,77 @@ static protected function HandleFacilityMissionExit ()
 	`SubmitGameState(NewGameState);
 }
 
+static protected function DisableChosenSurveillance ()
+{
+	local XComGameState_MissionSite MissionState;
+	local XComGameState_BattleData BattleData;
+	local XComGameStateHistory History;
+	
+	local XComGameState_HeadquartersAlien AlienHQ;
+	local XComGameState_AdventChosen ChosenState;
+	local StateObjectReference ChosenRef;
+
+	local XComGameState_ActivityChain ChainState;
+	local XComGameState NewGameState;
+	local XComGameState_Complication ComplicationState;
+	local StateObjectReference ComplicationRef;
+	local bool bClearComplication;
+
+	History = `XCOMHISTORY;
+	BattleData = XComGameState_BattleData(History.GetSingleGameStateObjectForClass(class'XComGameState_BattleData'));
+	MissionState = XComGameState_MissionSite(History.GetGameStateForObjectID(BattleData.m_iMissionID));
+
+	`CI_Log("DisableChosenSurveillance called");
+
+	// Do nothing if we came back from some other mission
+	if (MissionState.Source != 'MissionSource_ChosenStronghold') return;
+	
+	AlienHQ = XComGameState_HeadquartersAlien(`XCOMHISTORY.GetSingleGameStateObjectForClass(class'XComGameState_HeadquartersAlien'));
+	
+	`CI_Log("DisableChosenSurveillance exited stronghold");
+
+	// Do nothing if any chosen is still alive and kicking
+	foreach AlienHQ.AdventChosen(ChosenRef)
+	{
+		ChosenState = XComGameState_AdventChosen(`XCOMHISTORY.GetGameStateForObjectID(ChosenRef.ObjectID));
+			
+		if (!ChosenState.bDefeated)
+		{
+			return;
+		}
+	}
+	
+	`CI_Log("DisableChosenSurveillance chosen are dead");
+
+	foreach History.IterateByClassType(class'XComGameState_ActivityChain', ChainState)
+	{
+		if (ChainState.bEnded) continue;
+		if (ChainState.ComplicationRefs.Length == 0) continue;
+
+		foreach ChainState.ComplicationRefs(ComplicationRef)
+		{
+			ComplicationState = XComGameState_Complication(History.GetGameStateForObjectID(ComplicationRef.ObjectID));
+			if (ComplicationState.GetMyTemplateName() == 'Complication_ChosenSurveillance')
+			{
+				bClearComplication = true;
+				break;
+			}
+		}
+
+		if (!bClearComplication) continue;
+		
+		`CI_Log("DisableChosenSurveillance on chain" @ ChainState.GetMyTemplateName());
+
+		if (NewGameState == none) NewGameState = class'XComGameStateContext_ChangeContainer'.static.CreateChangeState("CI: PostMissionDisableSurviellance");
+
+		ChainState = XComGameState_ActivityChain(NewGameState.ModifyStateObject(class'XComGameState_ActivityChain', ChainState.ObjectID));
+		ChainState.ComplicationRefs.RemoveItem(ComplicationRef);
+		NewGameState.RemoveStateObject(ComplicationRef.ObjectID);
```

Comment:
Please call `ComplicationState.RemoveComplication()` instead. This way prevents any cleanup logic from executing